src/main/java/pan/Pan.java

```diff
@@ -0,0 +1,87 @@
+package pan;
+
+import pan.enums.Commands;
+import pan.enums.TaskStatus;
+import pan.exceptions.InternalTestCases;
+import pan.exceptions.InvalidCommandException;
+
+/**
+ * Parser - Encapsulates logic with I/O operations in the CLI
+ * @author Jerome Goh
+ */
+public class Pan {
+    private Ui ui;
+    private TaskList taskList;
+
+    /**
+     * Constructs a Parser instance.
+     *
+     * @param ui Ui instance that parser depends on.
+     * @param taskList TaskList instance responsible for CRUD operations with chatbot's Tasks.
+     */
+    public Pan(Ui ui, TaskList taskList) {
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Represents the driver method to parse and validate user logic.
+     *
+     * @param instruction The input instruction from the user.
+     */
+    public String parseInput(String instruction) {
```

Comment:
Perhaps can use a switch case statement here as it seems that there are a lot of cases?